modules/gltf/src/lib/extensions/EXT_structural_metadata.ts

```diff
@@ -0,0 +1,376 @@
+  propertyTables: GLTF_EXT_structural_metadata_PropertyTable[],
+  schemaClassName: string
+): GLTF_EXT_structural_metadata_PropertyTable | null {
+  for (let i = 0, len = propertyTables.length; i < len; i++) {
+    const propertyTable = propertyTables[i];
+
+    if (propertyTable.class === schemaClassName) {
+      return propertyTable;
+    }
+  }
+
+  return null;
+}
+
+/**
+ * Getting string attributes from binary data.
+ * Spec - https://github.com/CesiumGS/3d-tiles/tree/main/specification/Metadata#strings
+ * @param data
+ * @param offsetsData
+ * @param stringsCount
+ */
+function getStringAttributes(
+  data: Uint8Array,
+  offsetsData: Uint8Array,
+  stringsCount: number
+): string[] {
+  const stringsArray: string[] = [];
+  const textDecoder = new TextDecoder('utf8');
+
+  let stringOffset = 0;
+  for (let index = 0; index < stringsCount; index++) {
+    const stringByteSize = offsetsData[index + 1] - offsetsData[index];
+
+    if (stringByteSize + stringOffset <= data.length) {
+      const stringData = data.subarray(stringOffset, stringByteSize + stringOffset);
+      const stringAttribute = textDecoder.decode(stringData);
+
+      stringsArray.push(stringAttribute);
+      stringOffset += stringByteSize;
+    }
+  }
+
+  return stringsArray;
+}
+
+function getEnumAttributes(
+  valuesData: Uint8Array,
+  schema: GLTF_EXT_structural_metadata_Schema,
+  schemaProperty: GLTF_EXT_structural_metadata_ClassProperty,
+  offsetsData: Uint8Array | undefined,
+  numberOfAttributeValues: number
+): string[] {
+  const attributeValueArray: string[] = [];
+
+  const enumType = schemaProperty.enumType;
+  if (!enumType) return [];
+
+  const enumEntry: GLTF_EXT_structural_metadata_Enum | undefined = schema.enums?.[enumType];
+  if (!enumEntry) return [];
+
+  for (let index = 0; index < numberOfAttributeValues; index++) {
+    const valuesIndex = offsetsData?.[index] || 0;
+    const valueOfEnum = valuesData[valuesIndex];
+    const enumValue = getEnumByValue(enumEntry, valueOfEnum);
+    // TODO: Is the name or the value of the Enum expected to be used here?
+    if (enumValue) {
+      attributeValueArray.push(enumValue.name);
+    }
+  }
+
+  return attributeValueArray;
+}
+
+function getEnumByValue(
+  enumEntry: GLTF_EXT_structural_metadata_Enum,
+  value: number
+): GLTF_EXT_structural_metadata_EnumValue | null {
+  for (const enumValue of enumEntry.values) {
+    if (enumValue.value === value) {
+      return enumValue;
+    }
+  }
+
+  return null;
+}
+
+function getScalarAttributes(
+  valuesData: Uint8Array,
+  schema: GLTF_EXT_structural_metadata_Schema,
+  schemaProperty: GLTF_EXT_structural_metadata_ClassProperty,
+  offsetsData: Uint8Array | undefined,
+  numberOfAttributeValues: number
+): number[] {
+  const attributeValueArray: number[] = [];
+
+  for (let index = 0; index < numberOfAttributeValues; index++) {
+    const valuesIndex = offsetsData?.[index] || 0;
+    const value = valuesData[valuesIndex];
```

Comment:
This only works for Uint8. You have to get data type from `schemaProperty.componentType`